File: 268.missing-number.0.go
/*
 * @lc app=leetcode id=268 lang=golang
 *
 * [268] Missing Number
 */
func missingNumber(nums []int) int {
    sum := 0
    for i, n := range nums {
        sum = sum ^ n ^ i
    }
    return sum ^ len(nums)
}

File: 268.missing-number.0.rs
/*
 * @lc app=leetcode id=268 lang=rust
 *
 * [268] Missing Number
 */
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let mut sum = 0;
        for (i, n) in nums.iter().enumerate() {
            sum ^= n ^ (i as i32);
        }
        return sum ^ (nums.len() as i32)
    }
}
